protocols/v2/subprotocols/template-distribution/src/set_new_prev_hash.rs

```diff
@@ -8,33 +8,56 @@ use binary_sv2::{Deserialize, Serialize, U256};
-/// [future_job] flag set, which is valid work based on the prev_hash contained in this message, the
-/// template_id field SHOULD be set to the job_id present in that NewTemplate message
-/// indicating the client MUST begin mining on that template as soon as possible.
+/// Message used by an upstream to indicate the latest block header hash to mine on.
+///
+/// Upon validating a new best block, the server **must** immediately send this message.
+///
+/// If a `NewMiningJob` or `NewExtendedMiningJob` message has previously been sent with the
+/// Future Job flag set, the [`SetNewPrevHash::template_id`] field **should** be set to the
+/// `job_id` present in that message.
+///
+/// Upon receiving this message, the downstream **must** begin mining on the template referenced If
+/// they already received a `NewMiningJob` or `NewExtendedMiningJob` message with the Future Job
+/// flag set.
+///
 /// TODO: Define how many previous works the client has to track (2? 3?), and require that the
 /// server reference one of those in SetNewPrevHash.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub struct SetNewPrevHash<'decoder> {
-    /// template_id referenced in a previous NewTemplate message.
+    /// Identifier of the template to mine on.
+    ///
+    /// This must be identical to previously sent [`crate::NewTemplate`] message.
     pub template_id: u64,
     /// Previous block’s hash, as it must appear in the next block’s header.
     #[cfg_attr(feature = "with_serde", serde(borrow))]
     pub prev_hash: U256<'decoder>,
-    /// The nTime field in the block header at which the client should start
-    /// (usually current time). This is NOT the minimum valid nTime value.
+    /// nTime field in the block header at which the client should start (usually current time).
+    ///
+    /// This is **not** the minimum valid nTime value.
     pub header_timestamp: u32,
     /// Block header field.
     pub n_bits: u32,
-    /// The maximum double-SHA256 hash value which would represent a valid
-    /// block. Note that this may be lower than the target implied by nBits in
-    /// several cases, including weak-block based block propagation.
+    /// The maximum double-SHA256 hash value which would represent a valid block. Note that this
+    /// may be lower than the target implied by nBits in several cases, including weak-block based
+    /// block propagation.
     #[cfg_attr(feature = "with_serde", serde(borrow))]
     pub target: U256<'decoder>,
 }
 
+/// Message used by an upstream to indicate the latest block header hash to mine on.
+///
+/// Upon validating a new best block, the server **must** immediately send this message.
+///
+/// If a `NewMiningJob` or `NewExtendedMiningJob` message has previously been sent with the
+/// Future Job flag set, the [`SetNewPrevHash::template_id`] field **should** be set to the
+/// `job_id` present in that message.
```

Comment:
```suggestion
/// `template_id` present in that message.
```

I think this should be the right description, I just asked on [discord](https://discord.com/channels/950687892169195530/1019861139573714994/1308745616943353898) to be 100% sure